callback functionality

mjirik opened the pull request:
Hello,
I added callback functionality. 
I also uploaded module to pypi and anaconda server. Installation of your module is easier  with `pip` or  `conda`.

Mira

mjirik committed:
pypi and conda distribution meta files

mjirik committed:
Bump version: 1.0.0 → 1.0.1

mjirik committed:
callback function

mjirik committed:
Bump version: 1.0.1 → 1.0.2

mjirik committed:
install notes

mjirik committed:
fixed join path on windows

mjirik committed:
test added

mjirik committed:
Bump version: 1.0.2 → 1.0.3

mjirik committed:
convert os.sep to '/'

mjirik committed:
Merge remote-tracking branch 'origin/master'

mjirik committed:
Bump version: 1.0.3 → 1.0.4

mjirik committed:
cleaning tests

mjirik committed:
Merge branch 'master' of github.com:mjirik/sftpsync-py

mjirik committed:
test different separator

mjirik committed:
Bump version: 1.0.4 → 1.0.5

mjirik committed:
full path test

mjirik committed:
Merge remote-tracking branch 'origin/master'

mjirik committed:
logging in test

mjirik committed:
logger message for joined paths

mjirik committed:
Bump version: 1.0.5 → 1.0.6

mjirik committed:
allways use slash instead of backslash

mjirik committed:
Bump version: 1.0.6 → 1.0.7

mjirik committed:
fix upload

mjirik committed:
logger make relative

mjirik committed:
check if relative is required

mjirik committed:
fix ident

mjirik committed:
Bump version: 1.0.7 → 1.0.8

bickerle commented on tests/sftpsync_test.py:
```diff
@@ -0,0 +1,34 @@
+    def test_connection(self):
+        from sftpsync import Sftp
+
+        sftp = Sftp('147.228.47.162', 'paul', 'P4ul')
```
Hi,
just some quick thoughts on that commit:
I wouldn't post my server access if I were you.
Besides, should you ever change that access the test would break. A cleaner way to do it would be to set up a local sftp-server with paramiko and generate some dummy files to sync.

Best regards
Sebastian

mjirik committed:
use sftpserver package to run local sftp server

Commit:
Fix exception SyntaxError

mjirik committed:
Merge pull request #1 from dotoscat/master

Fix exception SyntaxError

Commit:
Sync file permissions

Commit:
Close connections in unit tests

mjirik committed:
Merge pull request #2 from grantramsay/feat/sync-file-permissions

Feat/sync file permissions

Commit:
Do not set folder permissions for directories that are out of scope

Previously if the folder you were syncing was /home/jim/myproject it would try set permissions for /home, /home/jim etc.
Likely causing a permission error

mjirik committed:
Merge pull request #3 from grantramsay/fix/permission-denied

Fix/permission denied

mjirik committed:
Bump version: 1.0.8 → 1.0.9

mjirik committed:
Bump version: 1.0.9 → 1.0.10

mjirik committed:
fix windows path starting with 'C:/'

mjirik committed:
Bump version: 1.0.10 → 1.0.11

mjirik committed:
Bump version: 1.0.11 → 1.0.12

mjirik committed:
moved meta.yml

mjirik committed:
conda recipe update

mjirik committed:
Bump version: 1.0.12 → 1.0.13
